enterprise/models.py

```diff
@@ -73,9 +75,20 @@ class Meta(object):
     catalog = models.PositiveIntegerField(null=True, help_text=_("Course catalog for the Enterprise Customer."))
     active = models.BooleanField(default=True)
     history = HistoricalRecords()
-    identity_provider = models.SlugField(null=True, blank=True, default=None)
     site = models.ForeignKey(Site, related_name="enterprise_customers")
 
+    @property
+    def identity_provider(self):
+        """
+        Unique slug for the identity provider associated with this enterprise customer.
+
+        Returns `None` if enterprise customer does not have any identity provider.
+        """
+        try:
+            return self.enterprise_customer_identity_provider and self.enterprise_customer_identity_provider.provider_id
```

Comment:
@asadiqbal08 function `identity_provider` [here](https://github.com/edx/edx-enterprise/pull/16/files#diff-61ac02e9b82173a27f3bbf8a350dd5e2R99) is responsible for returning a string to be displayed on admin portal. We fallback to provider_id to cover the cases when `third_party_auth` app will not be present.